content/source/docs/plugin/framework/accessing-values.html.md

```diff
@@ -0,0 +1,261 @@
+---
+layout: "extend"
+page_title: "Plugin Development - Framework: Access State, Config, and Plan"
+description: |-
+  How to read values from state, config, and plan in the next-generation
+  Terraform provider development framework.
+---
+
+# Access State, Config, and Plan
+
+There are various points at which the provider needs access to the data from
+the practitioner's configuration, Terraform's state, or the plan that was
+generated. The same patterns are used for accessing this data, regardless of
+its source.
+
+The data is usually stored in a request object:
+
+```go
+func (m myResource) Create(ctx context.Context, req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse)
+```
+
+In this example, `req` holds the configuration and plan; there is no state
+value, as the resource doesn't exist in state yet (that's why it's being
+created).
+
+## Getting the Entire Configuration, Plan, or State
+
+One way to interact with configuration, plan, and state values is to convert
+the entire configuration, plan, or state into a Go type, then treat them as
+regular Go values. This has the benefit of letting the compiler check all your
+code that accesses values, but requires defining a type to contain the values.
+
+```go
+type resourceData struct {
+	Name types.String `tfsdk:"name"`
+	Age types.Number `tfsdk:"age"`
+	Registered types.Bool `tfsdk:"registered"`
+	Pets types.List `tfsdk:"pets"`
+	Tags types.Map `tfsdk:"tags"`
+	Address types.Object `tfsdk:"address"`
+}
+
+func (m myResource) Create(ctx context.Context, req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
+	var plan resourceData
+	err := req.Plan.Get(ctx, &plan)
+	if err != nil {
+		// TODO: handle error
+	}
+	// values can now be accessed like plan.Name.Value
+	// check if things are null with plan.Name.Null
+	// check if things are unknown with plan.Name.Unknown
+}
+```
+
+The configuration, plan, and state data is represented as an object, and
+accessed like an object. See the [conversion rules](#conversion-rules) for an
+explanation on how objects can be converted into Go types.
+
+Using the [`attr.Value` implementations](/docs/plugin/framework/types.html) can
+surface complexity that is unnecessary, however. In a create function,
+non-computed values are guaranteed not to be unknown, for example, so there's
+no need to worry about them. Likewise, a required value will never be null, and
```

Comment:
can we rewrite this sentence so it doesn't use a double negative "guaranteed not to be unknown"